WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +121,166 @@ private extension ReviewsDashboardCardViewModel {
+                    }
+                }
+                if stores.isAuthenticatedWithoutWPCom == false {
+                    group.addTask { [weak self] in
+                        do {
+                            try await self?.synchronizeNotifications()
+                        } catch {
+                            self?.syncingError = error
+                            DDLogError("⛔️ Dashboard (Reviews) — Error synchronizing notifications: \(error)")
+                        }
+                    }
+                }
             }
+        }
     }
 
-    /// Get products from storage if available, if not then fetch remotely.
+    /// update predicate and manually fetch so that new predicate applies.
     ///
-    @MainActor
-    private func fetchProducts(for productIDs: [Int64]) async throws {
-        try productsResultsController.performFetch()
-
-        // Check if all productIDs are available in storage
-        let allProductsAvailable = productIDs.allSatisfy { productID in
-            reviewProducts.contains { $0.productID == productID }
+    func updateProductsResultsController(for productIDs: [Int64]) {
+        productsResultsController.predicate = NSCompoundPredicate(
+            andPredicateWithSubpredicates: [sitePredicate(),
+                                            NSPredicate(format: "productID IN %@", productIDs)])
+        do {
+            try productsResultsController.performFetch()
         }
-
-        if !allProductsAvailable {
-            await loadReviewProducts(for: productIDs)
+        catch {
+            ServiceLocator.crashLogging.logError(error)
         }
     }
+}
 
+// MARK: Remote
+private extension ReviewsDashboardCardViewModel {
     @MainActor
-    func loadReviewProducts(for reviewProductIDs: [Int64]) async {
-        syncingData = true
-        syncingError = nil
-
-        do {
-            try await retrieveProducts(for: reviewProductIDs)
-        } catch {
-            syncingError = error
+    func synchronizeProductReviews (filter: ReviewsFilter? = nil) async throws -> [ProductReview] {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(ProductReviewAction.synchronizeProductReviews(siteID: siteID,
+                                                                          pageNumber: 1,
+                                                                          pageSize: Constants.numberOfItems,
+                                                                          status: currentFilter.productReviewStatus
+                                                                         ) { result in
+                switch result {
+                case .success(let reviews):
+                    continuation.resume(returning: reviews)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
```

Comment:
Nit: this can be simplified with `continuation.resume(with: result)`.

Reply:
Nice! Updated in https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/595c389cf45c1e207c5912f221b5173b79b6a692